pkgs/modules/replit/dotreplit-schema.json

```diff
@@ -0,0 +1,661 @@
+        "extensionId": {
+          "type": "string"
+        },
+        "buildCommand": {
+          "type": "string"
+        },
+        "outputDirectory": {
+          "type": "string"
+        },
+        "staticDirectory": {
+          "type": "string"
+        }
+      }
+    },
+
+    "GitHubImportConfig": {
+      "title": "GitHub Imports Configuration",
+      "description": "Template files that are necessary to use as a base for GitHub imports.",
+      "type": "object",
+      "properties": {
+        "requiredFiles": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+
+    "HintConfig": {
+      "title": "Hints configuration",
+      "description": "Hint that shows in the Console based on stderr.",
+      "type": "object",
+      "properties": {
+        "regex": {
+          "type": "string"
+        },
+        "message": {
+          "type": "string"
+        }
+      }
+    },
+
+    "HostingConfig": {
+      "title": "Static Repl Hosting Configuration",
+      "description": "Configure which route and directory to use for static Repl hosting.",
+      "type": "object",
+      "properties": {
+        "route": {
+          "type": "string"
+        },
+        "directory": {
+          "type": "string"
+        }
+      }
+    },
+
+    "InterpConfig": {
+      "title": "Interpreter Configuration",
+      "description": "Configure the interpreter to use in the Console.",
+      "type": "object",
+      "properties": {
+        "command": {
+          "$ref": "#/definitions/Command"
+        },
+        "prompt": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "noFileArgs": {
+          "type": "boolean"
+        }
+      }
+    },
+
+    "LanguageConfig": {
+      "title": "Language Configuration",
+      "description": "Per-language configurations.",
+      "type": "object",
+      "properties": {
+        "pattern": {
+          "type": "string"
+        },
+        "syntax": {
+          "type": "string"
+        },
+        "languageServer": {
+          "$ref": "#/definitions/LanguageServerConfig"
+        }
+      }
+    },
+
+    "LanguageServerConfig": {
+      "title": "Language Server Configuration",
+      "description": "LSP server configurations.",
+      "type": "object",
+      "properties": {
+        "start": {
+          "$ref": "#/definitions/Command"
+        },
+        "configuration": {
+          "type": "object",
+          "additionalProperties": true
+        },
+        "initializationOptions": {
+          "type": "object",
+          "additionalProperties": true
+        }
+      }
+    },
+
+    "NixConfig": {
+      "title": "Nix Configuration",
+      "description": "Configure Nix on Replit.",
+      "type": "object",
+      "properties": {
+        "channel": {
+          "enum": ["unstable", "stable-21_11", "stable-22_05", "stable-22_11", "stable-23_05"]
```

Comment:
darn, another place to update the channels.

Reply:
yeah - even though it doesn't _have_ to be specified as an enum, it'd be nice for users to be able to see exactly what options they have